docs/reference/bare_metal/c_small_libs.md

```diff
@@ -0,0 +1,57 @@
+# Using code optimised C libraries in Mbed OS
+
+Both the `ARM` and `GCC_ARM` toolchains support code optimised versions of their C standard libraries. You can build with them by creating a `mbed_app.json` with the following contents:
+
+ ```
+ {
```

Comment:
This example is incorrect as it doesn't switch off the RTOS, should we add the baremetal here?

Reply:
I wanted to separate the two concepts: RTOS/bare-metal and std/small libraries.
I can move the bare metal section at the top though to emphasize that small libraries should be used in bare metal mode (and probably not when using a RTOS)